### ISSUE TITLE
Running antivirus applications interfere with USVFS - document workarounds

### ISSUE THREAD

**Issue by Exergist** [Modorganizer2/modorganizer#111]
I'm trying to install and use SKSE64 with MO2 v2.0.8.3. I noticed I was having very intermittent success getting SKSE64 to load through MO2. So I uninstalled the SKSE64 files to see if I could make headway with just MO2 and vanilla Skyrim. Well the issue persisted. If I try to run the "Skyrim Special Edition" executable (SkyrimSE.exe) through MO2 it often goes to a black screen like it will load but then immediately drops back to the desktop (I don't even see the Bethesda logo). Please note that this is on Windows 10 (version 1607) with no mods installed on a fresh installation of Skyrim SE (through Steam). Steam is installed in my C:\Program Files (x86) folder but my SteamLibrary (where Skyrim SE files are) as well as MO2 are on a separate drive. 

I've noticed that if a wait a few minutes before trying again I seem to have more success. If I rename the vanilla "SkyrimSELauncher.exe" to "Launcher.exe" and "SkyrimSE.exe" to "SkyrimSELauncher.exe" I can keep opening the game via the (modified) "SkyrimSELauncher.exe" until I'm blue in the face, but sadly MO2 won't work this way. 

I should also note that I received the following errors from the Windows Event Log:

> dlls,processorArchitecture="x86",type="win32",version="1.0.0.0"
> E:\SteamLibrary\steamapps\common\Skyrim  Special Edition\Mod Organizer\helper.exe
> 
> dlls,processorArchitecture="x86",type="win32",version="1.0.0.0"
> E:\SteamLibrary\steamapps\common\Skyrim Special Edition\Mod  Organizer\nxmhandler.exe
> 
> dlls,processorArchitecture="x86",type="win32",version="1.0.0.0"
> E:\SteamLibrary\steamapps\common\Skyrim  Special Edition\Mod Organizer\ModOrganizer.exe

Clearly there are times when I can get MO2 to work, so this is very frustrating. 

If I exit MO2 and just load up SKSE64 (v2.0.4) via the skse64_loader.exe I get more intermittent success. I'd say the "waiting" period between crashes and success is lower when I directly try to start SKSE64 vs. trying to start it in MO2. So I figured it would be best if I try to explain the issue to you as well as the SKSE64 team. Still, the fact that I can't get the base game to reliably load through MO2 is discouraging. 

Any thoughts about whether some kind of workaround or mod is needed to unload ghost processes related to SkyrimSE for use with SKSE64 or MO2? Any thoughts or feedback you might provide would be most appreciated. 

Here is an example log file for when I made a single attempt to load SkyrimSE through MO2 and failed. 
[usvfs-2017-10-27_09-48-13.log](https://github.com/LePresidente/modorganizer/files/1421603/usvfs-2017-10-27_09-48-13.log)

Please let me know if I can provide additional information that might help. 

Thanks in advance!

**Comment by Silarn** [comment]
This is a known issue and had been fixed in development versions. You can work around the issue by launching skse64 via CMD or using a batch file.

**Comment by Exergist** [comment]
Thanks for the feedback @Silarn! I'm not sure I fully understand what you're suggesting since I thought that for mods managed with MO2 to work the game must be started through MO2 (either within the MO2 app itself or through a shortcut generated by the app)...so wouldn't launching SKSE64 outside of the MO2 environment not properly load the mods? Are you talking about launching the SKSE64 shortcut generated by MO2 via CMD or batch file? Also is this workaround needed for launching vanilla Skyrim SE (no SKSE64 or other mods) through MO2 reliably?

I've started over with a new installation of steam (C:\steam) as well as a fresh installation of Skyrim SE (C:\steam...\Skyrim Special Edition). I also removed all C++ Redistributable packages and directly installed the 2015 package from Microsoft. I have not modified (or cleaned) any of the game files and I've not enabled any administrative privileges or compatibility modes. I can successfully start the game through Steam or directly through the SkyrimSE.exe (both cases display the launcher that allows me to start the game by pressing "Play"). IIRC launching the SkyrimSELauncher.exe (or whatever the file name is) also brings up the launcher screen but pressing "Play" just hides and reshows the launcher panel which can be continuously looped through. This info is all for context since I better make sure the base Skyrim SE works before I start introducing SKSE64 and/or MO2 into the mix. 

I'll be performing additional testing hopefully later this week:
1. Changing compatibility modes and administrative privileges for the SkyrimSE.exe and/or SkyrimSELauncher.exe to see if I can get the game to start and bypass the launcher screen. 
2. Undoing any successful changes from item 1, installing SKSE64, and trying to get the game to launch with SKSE64 only (directly launching the SKSE64_loader). If I continue to have issues hopefully the SKSE64 logs will be informative, and I'll be requesting help on the web forums (Steam, Nexus, etc) and if necessary send the information to the SKSE64 team. Subsequent compatibility modes/admin privileges may be needed. 
3. Uninstalling SKSE64 and installing MO2 (latest version released by @LePresidente, which is currently v2.0.8.3b). It will be installed within C:\steam\...\Skyrim Special Edition\MO2. I want to see if I can get vanilla Skyrim SE to load reliably through MO2 (above-mentioned workarounds may be needed at this point). Subsequent compatibility modes/admin privileges may be needed. If MO2 attempts to install additional C++ Redistributables they will be removed and reinstalled directly from Microsoft. Assuming I don't need a workaround to run SkyrimSE from MO2 (@Silarn please verify!) I'll update this post with additional information if I have issues using just MO2 and vanilla Skyrim SE. 
4. Only if I can get Items 3 and 4 to reliably work will I try to run SKSE64 through MO2 (though above workarounds appear to be needed). Otherwise troubleshooting will become convoluted O_o

**Comment by Exergist** [comment]
Updates!

> Changing compatibility modes and administrative privileges for the SkyrimSE.exe and/or SkyrimSELauncher.exe to see if I can get the game to start and bypass the launcher screen.

1. No tweaks **_OR_** Run SkyrimSE.exe as admin **_OR_** Run SkyrimSELauncher.exe as admin **_OR_** Run SkyrimSE.exe in Windows 7 compatibility mode **_OR_** Run SkyrimSE.exe in Windows 7
compatibility mode **_AND_** as administrator **_OR_** Run SkyrimSE.exe **_AND_** SkyrimSELauncher.exe in Windows 7 compatibility mode **_AND_** as SkyrimSE.exe as administrator does the following:
   - No issues launching game from Steam (loads launcher, press "Play" to start)
   - No issues launching game from SkyrimSE.exe (loads launcher, press "Play" to start)
   - Loading SkyrimSELauncher.exe loads up the launcher, but pressing "Play" makes the launcher screen disappear briefly and then reappear. Pressing "Play" again loads the game.
2. Rename "SkyrimSELauncher.exe" to "Launcher.exe," rename "SkyrimSE.exe" to "SkyrimSELauncher.exe" and execute the renamed "SkyrimSELauncher.exe"
   - No error messages and Skyrim SE game loads without showing the Launcher screen (bypasses launcher)
   - Executing the "Launcher.exe" produces a continuous loop of displaying and hiding the launcher screen after pressing "Play"

> Getting SKSE64 to work with Vanilla Skyrim SE and no mod manager installed

I submitted my troubleshooting findings to the SKSE64 team and they suggested I check for any programs that were injecting themselves into all running applications on my machine. They suggested that MSI Afterburner, some virus scanners, some video card overlay software, etc. may cause problems. Sure enough I realized that I had RivaTuner running in association with MSI Afterburner. **When I turned RivaTuner's global Application Detection Level to "None" I stopped having launching issues with SKSE64.** IIRC I could have the SKSE64, SkyrimSE, and SkyrimSELauncher executables not running as admin and not running in compatibility mode and this still worked!

> Getting MO v2.0.8.3b to work with Vanilla Skyrim SE (and SKSE64)

So after learning about the issue with RivaTuner and SKSE64 I thought I'd give MO2 another go. Lo and behold the issue with loading both vanilla Skyrim SE as well as SKSE64 through MO2 disappeared!! **So it is good to know that overlay software like RivaTuner (and probably similar programs like FRAPS) may negatively impact MO2 operation.** I'm not sure how widely this information is known, but I'd like to humbly suggest to the MO2 development team that some verbiage be added somewhere to highlight this issue.
